Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2892 -- Upgrade CMS Design System to latest version (11.x)

Resolves [EREGCSC-2892](https://jiraent.cms.gov/browse/EREGCSC-2892)

**Description**

Upgrades CMS Design System from 2.x (several years old) to 11.x (new, current).  Several layers of changes occurred over the past nine major versions, so some tweaking/refactoring/rethinking was necessary.

**This pull request changes:**

- Upgrades CMS Design System npm package from 2.x to 11.x
- Upgrades FontAwesome from 5.x to 6.x
- Adds Bitter font face and changes OpenSans filenames to all lowercase so fonts load as expected locally and on the hosted environments
- Standardizes `spacer` CSS variables across application to use CMS Design System values; removes previously defined eRegs Design System spacers
- Tweaks around site to address CMS Design System changes, such as:
   - [`box-sizing` changes](https://design.cms.gov/foundation/css-normalization/?theme=core#global-defaults)
   - list (`ul`, `ol`) display value changes
   - [Header changes](https://design.cms.gov/foundation/typography/headings/?theme=core)
- Removes SCSS styles from Vue Single File Components and moves them to SCSS stylesheet (mainly `_pagination.scss`)
- Removes SCSS preprocessing from `eregs-vite` Vite dependency bundling
- Addresses all warnings and deprecation messages in console when compiling Sass to CSS
- Tweaks print styling to hide Sign In header component
- Removes some unused frontend dependencies to partly satisfy [EREGCSC-2758](https://jiraent.cms.gov/browse/EREGCSC-2758). Examples:
   - `vue-template-compiler` v2.7 -- this is no longer necessary since we've upgraded Vue to v3

**Steps to manually verify this change:**

1. Green check marks
2. Site looks the same as `prod` except where indicated above
3. Static assets are compiled and built properly locally wen running `make local` and `make watch-all`
